applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -37,13 +37,29 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_USB_STATE_LOG_LEVEL);
 #define CONFIG_USB_HID_PROTOCOL_CODE -1
 #endif
 
+#define USB_HID_SELECTIVE_SUB	\
+	IS_ENABLED(CONFIG_DESKTOP_USB_SELECTIVE_HID_SUBSCRIPTION)
+
+#if USB_HID_SELECTIVE_SUB
+  #include "usb_state_def.h"
```

Comment:
I prefer no to duplicate configuration. If there is `usb_state_def.h` let's keep thing there and define it for all boards.
Adding config option may look easier but read again and see how many cases in the source code you have added.

Reply:
I think that, since most of nrf_desktop devices do not use this feature, leaving a default option would be a good idea. Also I would like to prevent users from defining this for dongles (that could lead to strange behaviors, since we do not know in which order devices will bond).

Reply:
I understand but don't agree.
There are several cons:
- you add several cases in the code, which complicates the code
- you don't know if this feature would not be useful for dongles - we don't use it but somebody may be willing to

Defining config file that makes all instances to accept everything would be easy to interpret.

Reply:
move this up so it is grouped with other headers

Reply:
The problem is that if someone uses this features for dongles he will encounter strange behaviour.
Depending of order in which peripherals bonds, other reports may be dropped. Handling this properly would require ensuring a predefined order of subscribers in hid_forward module and adding this could actually be a bit complicated.

Reply:
> The problem is the fact, that if someone uses this feature for a dongle, he will encounter strange behaviour:
> Depending on order in which peripherals bonds, various HID reports may be dropped. Handling this properly would require ensuring a predefined order of subscribers in hid_forward module and adding this could actually be a bit complicated.

I guess you are referring to "you don't know if this feature would not be useful for dongles - we don't use it but somebody may be willing to". If so, that is true but we don't add this now so we don't need to worry.

Reply:
I added an assert to prevent user from turning this on by the accident.